commerce_coordinator/apps/lms/tasks.py

```diff
@@ -147,11 +146,19 @@ def fulfill_order_placed_send_enroll_in_course_task(
 
     # Updating the order version and stateID after the transition to 'Fulfillment Failure'
     if self.request.retries > 0:
+        logger.warning(f"{tag} "
+                       f"Task retry count# {self.request.retries} for CT order ID {order_id}.")
         client = CommercetoolsAPIClient()
         # A retry means the current line item state on the order would be a failure state
-        line_item_state_id = client.get_state_by_key(TwoUKeys.FAILURE_FULFILMENT_STATE).id
-        start_time = datetime.now()
-        order_version = client.get_order_by_id(order_id).version
+        try:
+            line_item_state_id = client.get_state_by_key(TwoUKeys.FAILURE_FULFILMENT_STATE).id
+            start_time = datetime.now()
+            order_version = client.get_order_by_id(order_id).version
+        except CommercetoolsError as err:
+            logger.error(f'[CT-{tag}] Order not found: {order_id} with CT error {err}, {err.errors}'
+                         f', message id: {message_id}')
+            return False
+
```

Comment:
Revert these changes please. we need this exception to be raised unhandled in order for retry logic to work.

Adding in the retry criteria above is just enough.